controllers/imageclusterinstall_controller.go

```diff
@@ -902,48 +885,72 @@ func (r *ImageClusterInstallReconciler) writeClusterInfo(ctx context.Context, lo
 	if err != nil {
 		return err
 	}
-	var clusterID string
-	if existingInfo != nil && existingInfo.ClusterID != "" {
-		clusterID = existingInfo.ClusterID
-	} else if ici.Spec.ClusterMetadata != nil {
-		clusterID = ici.Spec.ClusterMetadata.ClusterID
-	} else {
-		clusterID = uuid.New().String()
-		log.Infof("created new cluster ID %s", clusterID)
+	if ici.Spec.ClusterMetadata == nil || ici.Spec.ClusterMetadata.ClusterID == "" || ici.Spec.ClusterMetadata.InfraID == "" {
+		return fmt.Errorf("clusterID and infraID are missing from cluster metadata")
 	}
 
-	var infraID string
-	if existingInfo != nil && existingInfo.InfraID != "" {
-		infraID = existingInfo.InfraID
-	} else if ici.Spec.ClusterMetadata != nil {
-		infraID = ici.Spec.ClusterMetadata.InfraID
-	} else {
-		infraID = generateInfraID(cd.Spec.ClusterName)
-		log.Infof("created new infra ID %s", infraID)
+	config := &imagebased.Config{
+		TypeMeta:             metav1.TypeMeta{APIVersion: imagebased.ImageBasedConfigVersion},
+		Hostname:             ici.Spec.Hostname,
+		ReleaseRegistry:      releaseRegistry,
+		AdditionalNTPSources: ici.Spec.AdditionalNTPSources,
+		NetworkConfig:        &aiv1beta1.NetConfig{Raw: []byte(nmstate)},
+		ClusterID:            ici.Spec.ClusterMetadata.ClusterID,
+		InfraID:              ici.Spec.ClusterMetadata.InfraID,
 	}
 
-	info := lca_api.SeedReconfiguration{
-		APIVersion:                lca_api.SeedReconfigurationVersion,
-		BaseDomain:                cd.Spec.BaseDomain,
-		ClusterName:               cd.Spec.ClusterName,
-		ClusterID:                 clusterID,
-		InfraID:                   infraID,
-		MachineNetwork:            ici.Spec.MachineNetwork,
-		SSHKey:                    ici.Spec.SSHKey,
-		ReleaseRegistry:           releaseRegistry,
-		Hostname:                  ici.Spec.Hostname,
-		KubeconfigCryptoRetention: KubeconfigCryptoRetention,
-		PullSecret:                psData,
-		RawNMStateConfig:          nmstate,
-		KubeadminPasswordHash:     kubeadminPasswordHash,
-		Proxy:                     r.proxy(ici.Spec.Proxy),
+	data, err := json.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("failed to marshal cluster info: %w", err)
+	}
+	if err := os.WriteFile(file, data, 0644); err != nil {
+		return fmt.Errorf("failed to write cluster info: %w", err)
 	}
 
+	return nil
+}
+
+func (r *ImageClusterInstallReconciler) writeInstallConfig(
```

Comment:
Make this a static method, it doesn't require the ImageClusterInstallReconciler.
Also consider moving it to the installer package (seems to belong there anyway)